runtime/autoload/nvimdiagnostic/python.vim

```diff
@@ -0,0 +1,384 @@
+function! s:version_info(python) abort
+  let pypi_version = s:latest_pypi_version()
+  let python_version = s:trim(system(
+        \ printf('"%s" -c "import sys; print(''.''.join(str(x) '
+        \ . 'for x in sys.version_info[:3]))"', a:python)))
+  if empty(python_version)
+    let python_version = 'unknown'
+  endif
+  
+  let nvim_path = s:trim(system(printf('"%s" -c "import sys, neovim;'
+        \ . 'print(neovim.__file__)" 2>/dev/null', a:python)))
+  if empty(nvim_path)
+    return [python_version, 'not found', pypi_version, 'unknown']
+  endif
+
+  let nvim_version = 'unknown'
+  let base = fnamemodify(nvim_path, ':h')
+  for meta in glob(base.'-*/METADATA', 1, 1) + glob(base.'-*/PKG-INFO', 1, 1)
+    for meta_line in readfile(meta)
+      if meta_line =~# '^Version:'
+        let nvim_version = matchstr(meta_line, '^Version: \zs\S\+')
+      endif
+    endfor
+  endfor
+
+  let version_status = 'unknown'
+  if nvim_version != 'unknown' && pypi_version != 'unknown'
+    if s:version_cmp(nvim_version, pypi_version) == -1
+      let version_status = 'outdated'
+    else
+      let version_status = 'up to date'
+    endif
+  endif
+
+  return [python_version, nvim_version, pypi_version, version_status]
+endfunction
+
+
+" Check the Python interpreter's usability.
+function! s:check_bin(bin, notes) abort
+  if !filereadable(a:bin)
+    call add(a:notes, printf('Error: "%s" was not found.', a:bin))
+    return 0
+  elseif executable(a:bin) != 1
+    call add(a:notes, printf('Error: "%s" is not executable.', a:bin))
+    return 0
+  endif
+  return 1
+endfunction
+
+
+" Text wrapping that returns a list of lines
+function! s:textwrap(text, width) abort
+  let pattern = '.*\%(\s\+\|\_$\)\zs\%<'.a:width.'c'
+  return map(split(a:text, pattern), 's:trim(v:val)')
+endfunction
+
+
+" Echo wrapped notes
+function! s:echo_notes(notes) abort
+  if empty(a:notes)
+    return
+  endif
+
+  echo '  Messages:'
+  for msg in a:notes
+    let msg_lines = s:textwrap(msg, 74)
+    echo '    *' msg_lines[0]
+    if len(msg_lines) > 1
+      echo join(map(msg_lines[1:], '"      ".v:val'), "\n")
+    endif
+  endfor
+endfunction
+
+
+" Load the remote plugin manifest file and check for unregistered plugins
+function! s:diagnose_manifest() abort
+  echo 'Checking: Remote Plugins'
+  let existing_rplugins = {}
+
+  for item in remote#host#PluginsForHost('python')
+    let existing_rplugins[item.path] = 'python'
+  endfor
+
+  for item in remote#host#PluginsForHost('python3')
+    let existing_rplugins[item.path] = 'python3'
+  endfor
+
+  let require_update = 0
+  let notes = []
+
+  for path in map(split(&rtp, ','), 'resolve(v:val)')
+    let python_glob = glob(path.'/rplugin/python*', 1, 1)
+    if empty(python_glob)
+      continue
+    endif
+
+    let python_dir = python_glob[0]
+    let python_version = fnamemodify(python_dir, ':t')
+
+    for script in glob(python_dir.'/*.py', 1, 1)
+          \ + glob(python_dir.'/*/__init__.py', 1, 1)
```

Comment:
This is a useful diagonstic, but ideally nvim should never try to "interpret" the contents of a `rpython/LANG` directory, this should solely be the responsibility of LANG-host.
Though in the next rplugin refactor (I hope to look into it soon) the plan is to eliminate the need to run `UpdateRemotePlugins` at all, so let's keep this diagnostic for now until then.